apps/src/templates/SafeMarkdown.jsx

```diff
@@ -29,23 +30,58 @@ class SafeMarkdown extends React.Component {
     markdown: PropTypes.string.isRequired,
     openExternalLinksInNewTab: PropTypes.bool,
     className: PropTypes.string,
+    includeCodeCopyButton: PropTypes.bool,
   };
 
-  render() {
-    // We only open external links in a new tab if it's explicitly specified
-    // that we do so; this is absolutely not something we want to do as a
-    // general practice, but unfortunately there are some situations in which
-    // it is currently a requirement.
-    const parser = this.props.openExternalLinksInNewTab
-      ? markdownToReactExternalLinks
-      : markdownToReact;
+  static defaultProps = {
+    includeCodeCopyButton: false,
+  };
 
-    const rendered = parser.processSync(this.props.markdown).contents;
+  getParser() {
+    const components = {
+      // Use React component wrappers for Blockly XML elements to prevent
+      // React from warning us about invalid components.
+      ...blocklyComponentWrappers,
+      ...(this.props.includeCodeCopyButton ? {pre: CodeBlockWithCopy} : {}),
+    };
+
+    const parser = Parser.create()
```

Comment:
This logic matches what was stored in `markdownToReact` and `markdownToReactExternalLinks` before, but is just extracted into a `getParser` function that includes the `CodeBlockWithCopy` binding in the `components` if `includeCodeCopyButton` is true.